### ISSUE TITLE
深度手势用户的几点优化建议!顶上!

### ISSUE THREAD

**Issue by superDMS** [yingDev/WGestures2-mac-bugs#74]
您好:
为使得WG2更尽善尽美,本人作为一个深度手势软件使用用户从很多体验的角度出发来给些许关键建议:
1.希望可以设置轨迹起始最小距离,如gestruesign的轨迹设置菜单就非常详尽,WG2的设置过于简单,缺少1.全屏自动禁用手势选项 2.最小起始轨迹像素值选项  等等容错选项!
2.wg2命令虽有cmd,lua等高拓展性的形式存在,但仅限部分有代码功底的人,希望为正常用户再添加几项常用的功能,比如:打开/激活程序,可拖拽添加. 现在用最笨的cmd启动法,添加麻烦不说,还特别不好用,因为cmd逻辑是每次命令相当于无差别打开一次目标路径程序,,有些程序体验非常不好.....有的甚至还被单例限制.正确的逻辑应该是未打开时则打开目标程序,检测到已打开则激活窗口.可参考fastgesture.gestruesign等,好像wgesture1也有这个设定呀,wg2感觉抛弃正常用户偏向有编程功底的用户了,原生功能越来越少.当然,老大您也可多增加几段lua脚本演示来解决这个需求)!
3.即使禁用了边缘触发,  照样无法在边缘触发普通手势,估计即使边缘禁用了还是被已经禁用的边缘所限制!
4.cmd脚本打开路径和网址的速度执行效率过慢 要等上两三秒   lua打开网址就非常快  但lua无法打开路径  希望提升cmd explorer效率  或者增加 lua打开路径操作示范
5.手势列表可以分组折叠,但不能多选,希望可以多选操作,方便管理大量同类型的手势,增加全局手势列表按命令类型排列等管理手势列表的功能,.(分组和全局手势的关系有点模糊,我试了几次才发现原来分组里并不是全局手势的子集,是井水不犯河水的关系)
6.cmd命令加入剪切板内容占位符,如copytext.方便一些复制路径后的操作.

**Comment by nagaho5** [comment]
第三点+1，比如我把一个侧键设为全局播放/暂停键，后来偶然发现在边缘和角落都无法触发，只能四条边四个角各设置一次，非常繁琐。感觉这个问题非常影响使用体验，在WG1里并没有发现这样的问题，只要禁用边角就行。

关于第四点，我一开始用的时候测试过也是这样，明显cmd打开速度慢几秒，就全用lua了。不过lua是可以打开路径的，作者常用功能里写了，放在了注释里，你仔细看看，对不懂代码的普通用户不太友好倒是真的。毕竟发布还没多久，现在已经很好用了，希望能慢慢完善

**Comment by superDMS** [comment]
> 第三点+1，比如我把一个侧键设为全局播放/暂停键，后来偶然发现在边缘和角落都无法触发，只能四条边四个角各设置一次，非常繁琐。感觉这个问题非常影响使用体验，在WG1里并没有发现这样的问题，只要禁用边角就行。
> 
> 关于第四点，我一开始用的时候测试过也是这样，明显cmd打开速度慢几秒，就全用lua了。不过lua是可以打开路径的，作者常用功能里写了，放在了注释里，你仔细看看，对不懂代码的普通用户不太友好倒是真的。毕竟发布还没多久，现在已经很好用了，希望能慢慢完善

是的  现在非常好用了, 希望锦上添花就非常完美了!  另:怎么用lua打开路径? 如:WG.Cmd("E:\")或WG.Open("E:\"),都打不开路径,请指教.哪里出现问题了?

**Comment by yingDev** [comment]
3. 这倒是没考虑到的. 目前的情况就是: 但凡禁止触发, 就穿透.

即将发布的下个版本 2.7:
1. 增加了最小识别距离设置. 暂时没有加上 "全屏禁用" 选项, 还需要深思熟虑, 似乎放在 "触发方式" 上设置更合理.
2. 增加了 运行/激活应用程序, Web搜索
4 explorer 确实受系统限制很慢, 似乎没啥办法
5,6. 增加了 WG_SELECTED_TEXT / WG_CLIPBOARD_TEXT 变量

**Comment by superDMS** [comment]
> 1. 这倒是没考虑到的. 目前的情况就是: 但凡禁止触发, 就穿透.
> 
> 即将发布的下个版本 2.7:
> 
> 1. 增加了最小识别距离设置. 暂时没有加上 "全屏禁用" 选项, 还需要深思熟虑, 似乎放在 "触发方式" 上设置更合理.
> 2. 增加了 运行/激活应用程序, Web搜索
>    4 explorer 确实受系统限制很慢, 似乎没啥办法
>    5,6. 增加了 WG_SELECTED_TEXT / WG_CLIPBOARD_TEXT 变量

嗯!非常好!慢慢完善!

**Comment by yingDev** [comment]
另外, 用 start 替换 explorer 会很快

**Comment by superDMS** [comment]
> 另外, 用 start 替换 explorer 会很快

!是的! 实测有效 !!start既能打开目录也能打开网址!   那不用explorer了!可以把有关explorer的都去掉了  start更好用! 之前舍近求远了,为了性能还去用lua.....我测试过了  lua都打开网址没start快,,,,,除非需要加入逻辑关系,才用lua,,,简单的打开完全不需要!

**Comment by superDMS** [comment]
另外:start打开正常网址没问题,但打开包含utf-8的链接就不行,如:https://www.baidu.com/s?ie=UTF-8&wd=%E5%A4%A9%E6%B0%94

**Comment by yingDev** [comment]
@superDMS  并不是这个原因, 而是 cmd 中, % 和 & 都是特殊字符. 所以, 可以手动编辑下使用原文, 避免这些字符, 或者用 ^% ^& 进行转义: `start https://www.baidu.com/wd=天气`

**Comment by superDMS** [comment]
> @superDMS 并不是这个原因, 而是 cmd 中, % 和 & 都是特殊字符. 所以, 可以手动编辑下使用原文, 避免这些字符, 或者用 ^% ^& 进行转义: `start https://www.baidu.com/wd=天气`

明白!谢谢老大指教!新知识点get!

**Comment by zhangenming** [comment]
1.全屏自动禁用手势选项

很需要这个功能!

因为设置了触发角
玩LOL的时候总会误触
虽然已经把LOL设置到禁用名单里了

所以每次玩LOL都得退出WG  很难受
